src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/AuthorizationHeaderProvider.java

```diff
@@ -0,0 +1,17 @@
+package com.google.devtools.build.lib.bazel.repository.downloader;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AuthorizationHeaderProvider {
+
+    private static final Map<String, AuthorizationProtocol> TYPE_TO_PROTOCOL = new HashMap<>();
+
+    public AuthorizationHeaderProvider() {
+        TYPE_TO_PROTOCOL.put("github", new GithubAuthorizationProtocol());
```

Comment:
> > So the "github" authentication protocol is just sticking "token " in front of the password? And that is only necessary because the `.netrc` format does not allow to quote spaces?
> 
> That is right

Thanks for clarifying. That does not improve my opinion on the `.netrc` format, but if it is popular, we'll have to support it.

Reply:
> Unfortunately, github uses a non-standard Authorization protocol, so this is specific for github.
> 
> [OAuth2](https://www.oauth.com/oauth2-servers/making-authenticated-requests/) states:
> `Authorization: Bearer <token>`
> 
> [Github](https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/) states:
> `Authorization: token <token>`

OK, so we have to special case it, unless we define a parametric `WithPrefixAuthenticationProtocol`.

> It is possible in future to add generic protocols, such as OAuth2, Basic authentication, etc.

I would appreciate if `bazel` wouldn't add authentication for GitHub only.

Reply:
on it